tests/sentry/incidents/serializers/test_workflow_engine_data_condition.py

```diff
@@ -0,0 +1,225 @@
+            "inputChannelId": None,
+            "integrationId": None,
+            "sentryAppId": None,
+            "dateCreated": self.critical_trigger_action.date_added,
+            "desc": f"Send a notification to {self.user.email}",
+            "priority": self.critical_action.data.get("priority"),
+        }
+        expected_actions.append(actions_2)
+        expected_trigger = self.expected_trigger.copy()
+        expected_trigger["actions"] = expected_actions
+        assert serialized_data_condition == expected_trigger
+
+    def test_comparison_delta(self) -> None:
+        comparison_delta_rule = self.create_alert_rule(comparison_delta=60)
+        comparison_delta_trigger = self.create_alert_rule_trigger(
+            alert_rule=comparison_delta_rule, label="critical"
+        )
+        comparison_delta_trigger_action = self.create_alert_rule_trigger_action(
+            alert_rule_trigger=comparison_delta_trigger
+        )
+        _, _, _, detector, _, _, _, _ = migrate_alert_rule(comparison_delta_rule)
+        comparison_detector_trigger, _ = migrate_metric_data_conditions(comparison_delta_trigger)
+        resolve_trigger_data_condition, _ = migrate_resolve_threshold_data_conditions(
+            comparison_delta_rule
+        )
+        action, _, _ = migrate_metric_action(comparison_delta_trigger_action)
+
+        serialized_data_condition = serialize(
+            comparison_detector_trigger,
+            self.user,
+            WorkflowEngineDataConditionSerializer(),
+        )
+        expected_actions = self.expected_actions.copy()
+        expected_actions[0]["id"] = str(comparison_delta_trigger_action.id)
+        expected_actions[0]["alertRuleTriggerId"] = str(comparison_delta_trigger.id)
+
+        expected_trigger = self.expected_trigger.copy()
+        expected_trigger["actions"] = expected_actions
+        expected_trigger["alertThreshold"] = translate_data_condition_type(
+            detector.config.get("comparison_delta"),
+            comparison_detector_trigger.type,
+            comparison_detector_trigger.comparison,
+        )
+        expected_trigger["id"] = str(comparison_delta_trigger.id)
+        expected_trigger["alertRuleId"] = str(comparison_delta_rule.id)
+        assert serialized_data_condition == expected_trigger
+
+    def test_multiple_rules(self) -> None:
+        # create another comprehensive alert rule in the DB
+        alert_rule, critical_trigger, warning_trigger, critical_action, warning_action = (
+            self.create_rule_triggers_and_actions()
+        )
+        migrate_alert_rule(alert_rule)
+        critical_detector_trigger, _ = migrate_metric_data_conditions(critical_trigger)
+        warning_detector_trigger, _ = migrate_metric_data_conditions(warning_trigger)
+        migrate_resolve_threshold_data_conditions(alert_rule)
+        migrate_metric_action(critical_action)
+        migrate_metric_action(warning_action)
```

Comment:
nit: just to make the tests faster, i think we could just create these db models then ensure they're serialized as expected since tehse aren't making assertions against the created `alert_rule` models.

i think it's mostly the mixing the two different approaches;
1. if we want to have assertions against the alert rule to confirm the same response - we could serialize the alert rule and compare it to the serialized detector(+workflow)
2. we could only create the workflow models and then assert to the expected data (like this is currently doing for the assertions)

Reply:
I'm creating the alert rule models for the convenience of using the migration helpers to create all the same stuff we'd have after dual writing, it'd be a lot more manual set up if I didn't have them. I could rewrite the tests to assert they have the same payload as the alert rule serializer, but if I do that I'd probably want to rewrite the action serializer as well since the tests are written the same way, which I can do separately.